### PR TITLE
[Flax LLaMA] Fix attn dropout

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

Attention dropout was not activated in Flax LLaMA, despite it being so in PyTorch LLaMA: https://github.com/huggingface/transformers/blob/1a585c1222a56bcaecc070966d558d4a9d862e83/src/transformers/models/llama/modeling_llama.py#L430

=> this PR unifies the implementations across frameworks

### CODE REVIEW

File: src/transformers/models/llama/modeling_flax_llama.py [hunk 1]
```diff
@@ -289,6 +289,10 @@ def __call__(
         attention_mask = jnp.broadcast_to(jnp.expand_dims(attention_mask, axis=(-3, -2)), causal_mask.shape)
         attention_mask = combine_masks(attention_mask, causal_mask)
 
+        dropout_rng = None
+        if not deterministic and self.config.attention_dropout > 0.0:
+            dropout_rng = self.make_rng("dropout")
```
Comment (sanchit-gandhi) [inline src/transformers/models/llama/modeling_flax_llama.py:294]: The rng is already correctly initialised:
https://github.com/huggingface/transformers/blob/1a585c1222a56bcaecc070966d558d4a9d862e83/src/transformers/models/llama/modeling_flax_llama.py#L418-L419

and passed to the `__call__` of the flax nn module:
https://github.com/huggingface/transformers/blob/1a585c1222a56bcaecc070966d558d4a9d862e83/src/transformers/models/llama/modeling_flax_llama.py#L486

=> so everything is already set-up for us to use the dropout rng

**Review by ArthurZucker — APPROVED** [review]
great catch thanks!
